src/main/java/com/google/cloud/hadoop/io/bigquery/hive/HiveBigQueryMetaHook.java

```diff
@@ -0,0 +1,94 @@
+package com.google.cloud.hadoop.io.bigquery.hive;
+
+import com.google.common.base.Strings;
+import org.apache.hadoop.hive.metastore.HiveMetaHook;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class {@link HiveBigQueryMetaHook} can be used to validate and perform different actions during
+ * creation and dropping of Hive Table.
+ */
+public class HiveBigQueryMetaHook implements HiveMetaHook {
+
+  private static Logger LOG = LoggerFactory.getLogger(HiveBigQueryMetaHook.class);
+
+  /**
+   * Performs required validations prior to creating the table
+   *
+   * @param table Represents hive table object
+   * @throws MetaException
+   */
+  @Override
+  public void preCreateTable(Table table) throws MetaException {
+
+    // Check all mandatory table properties
+    for (String property : HiveBigQueryConstants.MANDATORY_TABLE_PROPERTIES) {
+      if (Strings.isNullOrEmpty(table.getParameters().get(property))) {
+        throw new MetaException(property + " table property cannot be empty.");
+      }
+    }
+
+    // Check compatibility with BigQuery features
+    // TODO: accept DATE column 1 level partitioning
+    if (table.getPartitionKeysSize() > 0) {
+      throw new MetaException("Creation of Partition table is not supported.");
+    }
+
+    if (table.getSd().getBucketColsSize() > 0) {
+      throw new MetaException("Creation of bucketed table is not supported");
+    }
+
+    if (table.getTableType().equals("EXTERNAL_TABLE")) {
```

Comment:
This actually does not matter. Both managed table and external table can have LOCATION clause to specify table at a different location. It is safe to remove this check. The location check is more important

Reply:
External table check has been removed